systemtests/bgp_test.go

```diff
@@ -0,0 +1,930 @@
+		for name := range containers {
+			go func(conts []*container) { endChan <- s.removeContainers(conts) }(containers[name])
+		}
+
+		for range containers {
+			<-endChan
+		}
+		allcontainers = nil
+	}
+	s.TearDownBgp(c)
+
+}
+
+/*TestBgpTriggerNetpluginRestart tests the following:
+1) Checks bgp peering restablished on netplugin restart
+2) Checks ping success between containers on netplugin restart
+3) Checks ping success between containers and non container workloads
+*/
+func (s *systemtestSuite) TestBgpTriggerNetpluginRestart(c *C) {
+
+	if s.fwdMode != "routing" {
+		c.Skip("Skipping test")
+	}
+
+	s.SetupBgp(c, false)
+	s.CheckBgpConnection(c)
+
+	var (
+		netNames      = []string{}
+		containers    = map[string][]*container{}
+		allcontainers = []*container{}
+	)
+
+	numContainer := s.containers
+	if numContainer < 3 {
+		numContainer = 3
+	}
+
+	for networkNum := 0; networkNum < numContainer/len(s.nodes); networkNum++ {
+		network := &client.Network{
+			TenantName:  "default",
+			NetworkName: fmt.Sprintf("net%d-%d", networkNum, 1),
+			Subnet:      fmt.Sprintf("10.1.%d.0/24", networkNum),
+			Gateway:     fmt.Sprintf("10.1.%d.254", networkNum),
+			PktTag:      1001 + networkNum,
+			Encap:       "vlan",
+		}
+
+		c.Assert(s.cli.NetworkPost(network), IsNil)
+		netNames = append(netNames, network.NetworkName)
+	}
+
+	for _, name := range netNames {
+		var err error
+		containers[name], err = s.runContainers(numContainer, false, name, nil)
+		c.Assert(err, IsNil)
+		allcontainers = append(allcontainers, containers[name]...)
+	}
+
+	time.Sleep(5 * time.Second)
+
+	endChan := make(chan error)
+
+	logrus.Infof("Running ping test")
+	c.Assert(s.pingTest(allcontainers), IsNil)
+
+	for _, node := range s.nodes {
+		c.Assert(node.stopNetplugin(), IsNil)
+		logrus.Info("Sleeping for a while to wait for netplugin's TTLs to expire")
+		time.Sleep(1 * time.Minute)
+		time.Sleep(30 * time.Second)
+		c.Assert(node.rotateLog("netplugin"), IsNil)
+		c.Assert(node.startNetplugin("-fwd-mode=routing"), IsNil)
+		c.Assert(node.runCommandUntilNoError("pgrep netplugin"), IsNil)
+		time.Sleep(15 * time.Second)
+		s.CheckBgpConnection(c)
+		time.Sleep(5 * time.Second)
+
+		logrus.Infof("Running ping test")
+		c.Assert(s.pingTest(allcontainers), IsNil)
+		time.Sleep(5 * time.Minute)
+	}
+
+	for name := range containers {
+		go func(conts []*container) { endChan <- s.removeContainers(conts) }(containers[name])
+	}
+
+	for range containers {
+		<-endChan
+	}
+
+	for _, netName := range netNames {
+		c.Assert(s.cli.NetworkDelete("default", netName), IsNil)
+	}
+
+	s.TearDownBgp(c)
+
+}
+
+/*TestBgpTriggerNetmasterRestart tests the following:
+1) Checks bgp peering on netmaster restart
+2) Checks ping success between containers on netmaster restart
+3) Checks ping success between containers and non container workloads
+*/
+/*
+func (s *systemtestSuite) TestBgpTriggerNetmasterRestart(c *C) {
+	if s.fwdMode != "routing" {
```

Comment:
Just for my understanding, why was this test case commented out?
Are you planning to add it later?